pkg/cloud/services/networking/securitygroups.go

```diff
@@ -168,7 +168,7 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 		controlPlaneRules = append(controlPlaneRules, getSGControlPlaneAdditionalPorts(openStackCluster.Spec.APIServerLoadBalancer.AdditionalPorts)...)
 	}
 
-	if openStackCluster.Spec.AllowAllInClusterTraffic {
+	if openStackCluster.Spec.ManagedSecurityGroups != nil && openStackCluster.Spec.ManagedSecurityGroups.AllowAllInClusterTraffic {
```

Comment:
It seems odd that we'd be in this function if ManagedSecurityGroups was nil. Also we checked it explicitly at the top of the function.

Still, belt and braces is no bad thing when dereferencing a pointer.

Reply:
yeah I thought the same but i'm trying to avoid pointer errors later.